misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/ProgramAnalyzerNodeVisitor.java

```diff
@@ -375,1133 +196,165 @@ public void visit(FunctionDefinitionNode functionDefinitionNode) {
-
-    }
-
-    @Override
-    public void visit(NamedArgMatchPatternNode namedArgMatchPatternNode) {
-
-    }
-
-    @Override
-    public void visit(MarkdownDocumentationNode markdownDocumentationNode) {
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
 
-    }
+        Optional<Symbol> symbol = model.symbol(variableDeclarationNode);
+        if (symbol.isEmpty()) {
+            return;
+        }
 
-    @Override
-    public void visit(MarkdownDocumentationLineNode markdownDocumentationLineNode) {
+        if (variableDeclarationNode.initializer().isEmpty()) {
+            return;
+        }
 
-    }
+        if (isClientObject(symbol.get())) {
+            resolveEndPoint(symbol.get(), variableDeclarationNode.initializer().get());
+        }
 
-    @Override
-    public void visit(MarkdownParameterDocumentationLineNode markdownParameterDocumentationLineNode) {
+        if (isClientObject(symbol.get()) || isRecordObject(symbol.get())) {
+            Optional<ExpressionNode> expressionNode = variableDeclarationNode.initializer();
 
-    }
+            if (expressionNode.isPresent() && expressionNode.get().kind() != SyntaxKind.SIMPLE_NAME_REFERENCE) {
+                this.registerVariableRef(getEndpointReference(symbol.get()), variableDeclarationNode);
+            }
+        } else if ((symbol.get().kind() == SymbolKind.VARIABLE) &&
+                (variableDeclarationNode.initializer().get().kind() == SyntaxKind.REMOTE_METHOD_CALL_ACTION)) {
+            this.registerVariableRef(variableDeclarationNode.typedBindingPattern().bindingPattern().
+                            location().lineRange().hashCode(),
+                    variableDeclarationNode);
+        }
 
-    @Override
-    public void visit(BallerinaNameReferenceNode ballerinaNameReferenceNode) {
+        if (variableDeclarationNode.typedBindingPattern().bindingPattern().kind() ==
+                SyntaxKind.CAPTURE_BINDING_PATTERN) {
 
-    }
+            Optional<Symbol> initSymbol = model.symbol(variableDeclarationNode.initializer().get());
 
-    @Override
-    public void visit(InlineCodeReferenceNode inlineCodeReferenceNode) {
+            Integer key = variableDeclarationNode.typedBindingPattern().bindingPattern().
+                    location().lineRange().hashCode();
+            SyntaxKind initializerKind = variableDeclarationNode.initializer().get().kind();
+            if (initSymbol.isPresent() && initializerKind == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+                putReference(key, getEndpointReference(initSymbol.get()));
+            } else if (initializerKind == SyntaxKind.NUMERIC_LITERAL) {
+                this.registerVariableRef(key, variableDeclarationNode);
+            } else if (initializerKind == SyntaxKind.STRING_LITERAL) {
+                this.registerVariableRef(key, variableDeclarationNode);
+            }
+        }
 
+        visitSyntaxNode(variableDeclarationNode);
```

Comment:
In the visitor pattern we don't call the `visitSyntaxNode` explicitly, in general. Use `accept` method instead